packages/grid/x-grid/src/tests/layout.XGrid.test.tsx

```diff
@@ -58,7 +58,26 @@ describe('<XGrid /> - Layout', () => {
         </div>,
       );
 
-      expect(document.querySelector(`.${className}`)).to.equal(container.firstChild.firstChild);
+      expect(container.firstChild.firstChild).to.have.class(className);
+      expect(container.firstChild.firstChild).to.have.class('MuiDataGrid-root');
+    });
+
+    it('applies the style to the root component', () => {
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <XGrid
+            {...baselineProps}
+            style={{
+              border: 0,
+            }}
+          />
+        </div>,
+      );
+
+      // @ts-expect-error need to migrate helpers to TypeScript
+      expect(document.querySelector('.MuiDataGrid-root')).toHaveInlineStyle({
```

Comment:
I didn't use the role="grid" because I trust more that we set the root class name on the root than the aria.